[Failing Test]: Many BigQuery direct read Python PostCommit failing possibly due to fastavro regression

### What happened?

Possibly due to 1.8.4 released today (Oct 3)

```
apache_beam/runners/common.py:1609: in handle_process_outputs
    for result in results:
apache_beam/io/gcp/bigquery.py:1312: in __next__
    return fastavro.schemaless_reader(self.bytes_reader, self.avro_schema)
fastavro/_read.pyx:1126: in fastavro._read.schemaless_reader
    ???
fastavro/_read.pyx:1153: in fastavro._read.schemaless_reader
    ???
fastavro/_read.pyx:743: in fastavro._read._read_data
    ???
fastavro/_read.pyx:616: in fastavro._read.read_record
    ???
fastavro/_read.pyx:735: in fastavro._read._read_data
    ???
fastavro/_read.pyx:526: in fastavro._read.read_union
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   EOFError

fastavro/_read.pyx:176: EOFError
```

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [x] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [x] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/python/apache_beam/io/gcp/bigquery.py
+++ b/sdks/python/apache_beam/io/gcp/bigquery.py
@@ -1310,7 +1310,7 @@ class _ReadReadRowsResponsesWithFastAvro():
   def __next__(self):
     try:
       return fastavro.schemaless_reader(self.bytes_reader, self.avro_schema)
-    except EOFError:
+    except (StopIteration, EOFError):
       self.read_rows_response = next(self.read_rows_iterator, None)
       if self.read_rows_response is not None:
         self.bytes_reader = io.BytesIO(

--- a/sdks/python/apache_beam/io/gcp/bigquery.py
+++ b/sdks/python/apache_beam/io/gcp/bigquery.py
@@ -1310,14 +1310,14 @@ class _ReadReadRowsResponsesWithFastAvro():
   def __next__(self):
     try:
       return fastavro.schemaless_reader(self.bytes_reader, self.avro_schema)
-    except StopIteration:
+    except EOFError:
       self.read_rows_response = next(self.read_rows_iterator, None)
       if self.read_rows_response is not None:
         self.bytes_reader = io.BytesIO(
             self.read_rows_response.avro_rows.serialized_binary_rows)
         return fastavro.schemaless_reader(self.bytes_reader, self.avro_schema)
       else:
-        raise StopIteration
+        raise EOFError
 
 
 @deprecated(since='2.11.0', current="WriteToBigQuery")

--- a/sdks/python/apache_beam/io/gcp/bigquery.py
+++ b/sdks/python/apache_beam/io/gcp/bigquery.py
@@ -1317,7 +1317,7 @@ class _ReadReadRowsResponsesWithFastAvro():
             self.read_rows_response.avro_rows.serialized_binary_rows)
         return fastavro.schemaless_reader(self.bytes_reader, self.avro_schema)
       else:
-        raise EOFError
+        raise StopIteration
 
 
 @deprecated(since='2.11.0', current="WriteToBigQuery")

